src/components/OverlayWrapper/OverlayWrapper.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import _ from 'lodash';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, getFirst, rejectTypes, omitProps } from '../../util/component-types';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+
+const cx = lucidClassNames.bind('&-OverlayWrapper');
+
+const {
+	bool,
+	node,
+	oneOf,
+	string,
+} = React.PropTypes;
+
+/**
+ *
+ * {"categories": ["utility"]}
+ *
+ * A wrapper with optional overlay to wrap content. `Overlay` is meant for overlaying an entire page, while this component is meant to wrap another component and cover its content.
+ *
+ */
+const OverlayWrapper = createClass({
+	displayName: 'OverlayWrapper',
+	propTypes: {
+		/**
+		 * Controls whether the message should be displayed over the wrapped content.
+		 */
+		isVisible: bool,
+		/**
+		 * Set this to `false` if you don't want the semi-transparent overlay over
+		 * the wrapped content.
+		 */
+		hasOverlay: bool,
+		/**
+		 * Class names that are appended to the defaults.
+		 */
+		className: string,
+		/**
+		 * Any valid React children.
+		 */
+		children: node,
+		/**
+		 * Style variations for the overlay behind the message.
+		 */
+		overlayKind: oneOf([
+			'light',
+			'dark',
+		]),
+		/**
+		 * *Child Element*
+		 *
+		 * The Message to display in the overlay.
+		 */
+		Message: node,
```

Comment:
I think I added this because it was failing the `'propNames in propTypes'` generic-test otherwise.